internal/handlers/pg/pgdb/query.go

```diff
@@ -268,46 +269,81 @@ func prepareWhereClause(sqlFilters *types.Document) (string, []any, error) {
 			}
 		}
 
-		// Handle _id with a simpler query, as it can't be an array
-		if k == "_id" {
-			switch v := v.(type) {
-			case *types.Document, *types.Array, types.Binary, bool, time.Time, types.NullType, types.Regex, types.Timestamp:
-				// type not supported for pushdown
-				// TODO $eq and $ne https://github.com/FerretDB/FerretDB/issues/1840
-				// TODO $gt and $lt https://github.com/FerretDB/FerretDB/issues/1875
-			case float64, string, types.ObjectID, int32, int64:
-				// Select if value under the key is equal to provided value.
-				sql := `(_jsonb->%[1]s)::jsonb = %[2]s`
-
-				// placeholder p.Next() returns SQL argument references such as $1, $2 to prevent SQL injections.
-				// placeholder $1 is used for field key,
-				// placeholder $2 is used for field value v.
-				filters = append(filters, fmt.Sprintf(sql, p.Next(), p.Next()))
-				args = append(args, k, string(must.NotFail(pjson.MarshalSingleValue(v))))
-
-			default:
-				panic(fmt.Sprintf("Unexpected type of value: %v", v))
-			}
+		// check if values are equal or if the left value contains the right one
+		eqOperator := "@>"
 
-			continue
+		if k == "_id" {
+			// check if values are equal
+			eqOperator = "="
```

Comment:
Why we need that?

Reply:
As `_id` cannot be an array, containment operator is not needed for equal operations on it.

Reply:
But is it worth making our code more complicated, given that index could be built for `@>` operators, so it would make sense to use only it?

Reply:
Previously we've used longer query for all other field, and simpler one for `_id`, so it made sense then. But yes, right now the difference is not big, and we can remove `_id` check for now. 

Maybe there are some differences in how effective Postgres evaluates both operators, but let's not focus on that right now. I will remove this check.